### ISSUE TITLE
Cannot establish connection between client and listener when using WebSockets

### ISSUE THREAD

**Issue by robreeves** [Azure/amqpnetlite#106]
When I use WebSockets instead of AMQP, my senderlink cannot connect to the listener. It gets rejected during the initial negotiation when the client sends the protocol. It is sending HTTP, but ProtocolHeader expects AMQP. The failure occurs at the protocol check in Amqp.Framing.ProtocolHeader.Create. Am I doing something wrong? Here is my example. If I use amqp instead of ws the example works fine.

```
    class Program
    {
        const string Address = "ws://localhost:8888/test/";
        const string MsgQueue = "queue";

        static void Main(string[] args)
        {
            Trace.TraceLevel = TraceLevel.Warning;
            Trace.TraceListener = (f, a) =>
            {
                Console.WriteLine(DateTime.Now.ToString("[hh:ss.fff]") + " " + string.Format(f, a));
            };

            var host = new ContainerHost(new Uri(Address));
            host.RegisterMessageProcessor(MsgQueue, new MessageProcessor());
            host.Open();

            var conn = new Connection(new Address(Address));
            var sess = new Session(conn);
            var sender = new SenderLink(sess, "client", MsgQueue);

            sender.Send(new Message("1"));

            Console.WriteLine("done");
            Console.ReadKey();
        }
    }

    class MessageProcessor : IMessageProcessor
    {

        public int Credit
        {
            get { return 1; }
        }

        public void Process(MessageContext messageContext)
        {
            Console.WriteLine(messageContext.Message.Body);
            messageContext.Complete();
        }
    }
```

**Comment by xinchen10** [comment]
@robreeves it was started on NETMF and got carried over to .NET and other platforms. I am doing some refactoring to deprecate the ctor on platforms where the factory is available.

**Comment by xinchen10** [comment]
fixed in 1.1.8
